mlflow/utils/databricks_utils.py

```diff
@@ -419,6 +430,31 @@ def _fail_malformed_databricks_auth(profile):
     )
 
 
+def _fail_model_serving_creds_env(exception):
+    raise MlflowException(
+        "Unable to read Oauth credentials from file mount for Databricks "
+        f"Model Serving dependency failed"
+    ) from exception
+
+
+# Helper function to attempt to read OAuth Token from
+# mounted file in Databricks Model Serving environment
+# constant defined outside function for testing override
+MODEL_DEPENDENCY_OAUTH_TOKEN_FILE_PATH = "/var/credentials-secret/model-dependencies-oauth-token"
+
+
+def _get_model_dependency_oauth_token(should_retry=True):
+    try:
+        with open(MODEL_DEPENDENCY_OAUTH_TOKEN_FILE_PATH) as f:
+            oauth_dict = json.load(f)
+            return oauth_dict["OAUTH_TOKEN"][0]["oauthTokenValue"]
+    except Exception as e:
```

Comment:
when do we need a retry? When `/var/credentials-secret/model-dependencies-oauth-token` is missing?

Reply:
ahh mainly this is in case of some weird edge case race condition where the serving scheduler is simultaneously updating the file at that point. Added a short wait before the retry.